fact-ebpf/process.h

```diff
@@ -108,11 +109,8 @@ __always_inline static int64_t process_fill(process_t* p) {
 
   unsigned long arg_start = BPF_CORE_READ(task, mm, arg_start);
   unsigned long arg_end = BPF_CORE_READ(task, mm, arg_end);
-  unsigned int len = arg_end - arg_start;
-  if (len > 4095) {
-    len = 4095;
-    p->args[4095] = '\0';  // Ensure empty string at end of buffer
-  }
+  unsigned int len = (arg_end - arg_start) & 0xFFF;
+  p->args[4095] = '\0';  // Ensure string termination at end of buffer
```

Comment:
should this be `p->args[len] = '\0'`? or should the string already be null terminated and this is a safety against overrunning the buffer?

Reply:
The arguments in the kernel are stored as a buffer of strings, so if the length is shorter than 4K it will already be a set of null terminated strings, if it's longer, this will make sure it has the termination.